### ISSUE TITLE
Implement JobApplication state change to 'not_accepted'

### ISSUE THREAD

**Issue by patmbolger** [AgileVentures/MetPlus_tracker#374]
As a user with appropriate authorization
I want to be able to mark a job application as 'not_accept'

**Comment by patmbolger** [comment]
See JobApplication states as defined in this page:  http://www.agileventures.org/projects/metplus/documents/system-behavior

Note that this state change can be initiated via:

1. The user setting another job application (for the same job) to 'accepted' status, or,
2. The (authorized) user explicitly marking a job application as 'not accepted'.

This story includes:

- [ ] Allow user to set job application status to 'not_accepted' (authorized users will be granted access via Pundit, so no need to limit access in this story)

- [ ] When job application is mark not_accepted (regardless of state change mechanism), notify job seeker's primary job developer (realtime popup and email) (that is, the job developer associated with the job seeker who applied to the job)

- [ ] When job application is mark not_accepted (regardless of state change mechanism), notify job seeker (realtime popup and email)

**Comment by patmbolger** [comment]
Action item from client meeting (http://www.agileventures.org/projects/metplus/documents/meeting-22-07-15-2016):

"When a user is not accepted for a job, the company person should give a reason" -

The client wants the user to specify a reason for not accepting the application.  We talked about using 1) text entry and 2) selecting a code specifying a pre-defined reason.  We decided that initially we will just use a text entry field.  Later, we might also add codes that can be selected.

Note: 

1. **entry of text into the field is mandatory** - the model needs to validate this.

2. If the job application is marked as 'not_accepted' because another application was 'accepted' (state change cause #1 above) then the text field should contain 'Another job application was accepted.'
